airbyte-integrations/base-java/src/main/java/io/airbyte/integrations/javabase/Destination.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2020 Airbyte
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package io.airbyte.integrations.javabase;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import io.airbyte.config.DestinationConnectionSpecification;
+import io.airbyte.config.Schema;
+import io.airbyte.config.StandardCheckConnectionOutput;
+import io.airbyte.config.StandardDiscoverSchemaOutput;
+import io.airbyte.singer.SingerMessage;
+import java.io.IOException;
+
+// todo (cgardens) - share common parts of this interface with source.
+public interface Destination {
+
+  DestinationConnectionSpecification spec() throws IOException;
+
+  StandardCheckConnectionOutput check(JsonNode config) throws IOException;
+
+  StandardDiscoverSchemaOutput discover(JsonNode config) throws IOException;
+
+  DestinationConsumer<SingerMessage> write(JsonNode config, Schema schema) throws IOException;
```

Comment:
maybe `getWriter`?

Reply:
isn't this the equivalent of how you're doing it in the python version?

```
    # Iterator<AirbyteMessage>
    def read(self, config_object, rendered_config_path, state=None) -> Generator[AirbyteMessage, None, None]:
        raise Exception("Not Implemented")
```

Reply:
This is a bit of a gray area. Does returning a lazy iterator actually count as reading the data?

`write` should be fine. I'll keep `read` in the python source version too even though it's slightly misleading too.

Matching the commands is nice, let's keep what you have here.